### PR TITLE
Tests for n<0

### LINKED ISSUE #94740: torch.diff of negative orders return the original array
### 🐛 Describe the bug

`torch.diff` accepts negative orders for the finite differences, and returns just the original array: 
```
In [133]: t
Out[133]: tensor([0, 1, 2, 3, 4, 5])

In [134]: torch.diff(t, n=-3)
Out[134]: tensor([0, 1, 2, 3, 4, 5])

In [135]: torch.diff(t, n=0)
Out[135]: tensor([0, 1, 2, 3, 4, 5])

In [139]: torch.diff(t, n=-1) is t
Out[139]: True
```

For reference, numpy allows `n=0` and errors out for `n < 0`:

```
In [136]: np.diff(t.numpy(), n=0)
Out[136]: array([0, 1, 2, 3, 4, 5])

In [137]: np.diff(t.numpy(), n=-3)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [137], in <cell line: 1>()
----> 1 np.diff(t.numpy(), n=-3)

File <__array_function__ internals>:200, in diff(*args, **kwargs)

File ~/miniconda3/envs/pytorch/lib/python3.9/site-packages/numpy/lib/function_base.py:1408, in diff(a, n, axis, prepend, append)
   1406     return a
   1407 if n < 0:
-> 1408     raise ValueError(
   1409         "order must be non-negative but got " + repr(n))
   1411 a = asanyarray(a)
   1412 nd = a.ndim

ValueError: order must be non-negative but got -3
```

### Versions

$ python collect_env.py
Collecting environment information...
PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.9.0 | packaged by conda-forge | (default, Nov 26 2020, 07:57:39)  [GCC 9.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-58-generic-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          8
On-line CPU(s) list:             0-7
Thread(s) per core:              2
Core(s) per socket:              4
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           142
Model name:                      Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz
Stepping:                        12
CPU MHz:                         1800.000
CPU max MHz:                     3900,0000
CPU min MHz:                     400,0000
BogoMIPS:                        3600.00
Virtualization:                  VT-x
L1d cache:                       128 KiB
L1i cache:                       128 KiB
L2 cache:                        1 MiB
L3 cache:                        6 MiB
NUMA node0 CPU(s):               0-7
Vulnerability Itlb multihit:     KVM: Mitigation: VMX disabled
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:          Mitigation; Enhanced IBRS
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:             Mitigation; Microcode
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust sgx bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==1.13.0
[conda] mkl                       2022.2.1         h84fe81f_16997    conda-forge
[conda] numpy                     1.24.1           py39h7360e5f_0    conda-forge
[conda] pytorch                   1.13.0          cpu_py39h3439074_0    conda-forge

### PR DESCRIPTION
**Pull request by Player256**
Fixes #94740

### CODE REVIEW

**Review by bdhirsh — APPROVED** [review]
thanks for the fix!

File: test/test_torch.py [hunk 1]
```diff
@@ -2631,6 +2631,8 @@ def test_diff(self, device, dtype):
                 RuntimeError, 'diff expects input to be at least one-dimensional'):
             scalar = torch.tensor(2, device=device, dtype=dtype)
             torch.diff(scalar)
+        with self.assertRaisesRegex(RuntimeError, 'order must be non-negative but got -1'):
```
Comment (bdhirsh) [inline test/test_torch.py:2634]: @Player256 this test is failing in CI, you need to update the error message:
```
  File "test_torch.py", line 2635, in test_diff
    torch.diff(t, n=-1)
AssertionError: "order must be non-negative but got -1" does not match "order must be non-negative-1"
```

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
I'm closing this as the issue was actually already fixed by https://github.com/pytorch/pytorch/pull/95144

**Pull request closed without merging**